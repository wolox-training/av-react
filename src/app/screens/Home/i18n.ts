import i18next from 'i18next';

i18next.addResources('es', 'Home', {
  logoAlt: 'Logo',
  welcome: 'Â¡Bienvenido al training!'
});
