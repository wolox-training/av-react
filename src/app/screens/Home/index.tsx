import React from 'react';

import styles from './styles.module.scss';
import Appbar from './components/Appbar';
import BookList from './components/BookList';

function Home() {
  return (
    <div className={`${styles.home} column`}>
      <Appbar />
      <BookList />
    </div>
  );
}

export default Home;
