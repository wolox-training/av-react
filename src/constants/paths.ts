export const PATHS = {
  login: '/',
  signup: '/sign_up'
};
