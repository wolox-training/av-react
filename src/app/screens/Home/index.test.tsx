import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import { PATHS } from '~constants/paths';

import Home from './';

const history = createMemoryHistory();
const pushSpy = jest.spyOn(history, 'push');

describe('Home test', () => {
  let container: Element | null = null;

  beforeEach(() => {
    const { container: renderContainer } = render(
      <Router history={history}>
        <Home />
      </Router>
    );
    container = renderContainer;
  });

  test('Should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  test('Should render appbar content', () => {
    expect(screen.getByAltText('Home:logoAlt')).toBeInTheDocument();
    expect(screen.getByText('Home:logout')).toBeInTheDocument();
  });

  test('Should redirect to login', async () => {
    userEvent.click(screen.getByText('Home:logout'));
    await waitFor(() => expect(pushSpy).toHaveBeenCalledWith(PATHS.login));
  });
});
