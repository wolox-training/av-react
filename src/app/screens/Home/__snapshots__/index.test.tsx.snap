// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test Should match snapshot 1`] = `
<div>
  <div
    class="home column"
  >
    <nav
      class="appbar row space-around middle"
    >
      <img
        alt="Home:logoAlt"
        class="logo"
        src="wolox-logo.png"
      />
      <h2
        class="logout"
      >
        Home:logout
      </h2>
    </nav>
    <div
      class="booksList"
    >
      <div
        class="sk-fade-in-half-second sk-spinner sk-three-bounce"
        color="#000"
        style="color: rgb(0, 0, 0);"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
